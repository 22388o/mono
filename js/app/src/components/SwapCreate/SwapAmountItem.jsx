import React, { useCallback, useSyncExternalStore } from 'react'
import styles from '../../styles/SwapCreate.module.css'
import { Grid, Button } from '@mui/material'
import { walletStore } from '../../syncstore/walletstore'

export const SwapAmountItem = ({ assetId, amount, className, onAmountChange, unitPrice, availQty }) => {
  const globalWallet = useSyncExternalStore(walletStore.subscribe, () => walletStore.currentState)
  const asset = globalWallet.assets[assetId]

  const onKeyDown = useCallback((e) => {
    if (e.keyCode === 109) {
      e.preventDefault()
    }
  }, [])

  return (
    <>
      <Grid container direction='row' className={className} sx={{ height:'48px' }}>
        <Grid item xs={7} container direction='column' textAlign='left'>
          <input
            className={`${styles['qty-input']} qty-input`}
            placeholder={availQty}
            type='number'
            value={(amount === 0) ? '' : amount}
            onChange={onAmountChange}
            onKeyDown={onKeyDown}
          />
          {unitPrice * amount > 0 ? <span className={styles.prices}>${unitPrice * amount}</span> : ''}
        </Grid>
        <Grid item xs={5} textAlign='right' sx={{display:'flex',alignItems:'center',justifyContent:'flex-end'}}>
          <Button className={`${styles['coin-select']} coin-select`} onClick={() => {}}><img src={asset.img_url} />{asset.type.split('-')[0]}</Button>
        </Grid>
      </Grid>
    </>
  )
}
