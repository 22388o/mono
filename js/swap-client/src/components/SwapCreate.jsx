import { useState } from "react";
import 'semantic-ui-css/semantic.min.css';
import { Button, Card, Form } from 'semantic-ui-react';
import { setSwapId, setSwapHash, setSecretSeekerId, setSecretHolderId, setSecret, setBase, setQuote } from "../slices/swapSlice";
import { useAppDispatch } from "../hooks";
import { fetchSwapCreate } from "../utils/apis";
import { useNavigate } from "react-router-dom";
import { addSwapItem } from "../slices/historySlice";

export const SwapCreate = () => {
  const navigate = useNavigate();
	const dispatch = useAppDispatch();
  const [baseQuantity, setBaseQuantity] = useState(10000)
  const [quoteQuantity, setQuoteQuantity] = useState(30000)
  const [pressed, setPressed] = useState(false);

  const onCreateSwap = async () => {
    fetchSwapCreate({baseQuantity, quoteQuantity})
    .then(res => {
      console.log(res);
      return res.json();
    })
    .then(data => {
      console.log(data.swap.id)
      console.log(`${JSON.stringify(data)}`)
      dispatch(setBase(baseQuantity));
      dispatch(setQuote(quoteQuantity));
      dispatch(setSwapId(data.swap.id));
      dispatch(setSecretSeekerId(data.swap.secretSeeker.id));
      dispatch(setSecretHolderId(data.swap.secretHolder.id));
      dispatch(setSecret(data.swapSecret));
      dispatch(setSwapHash(data.swap.secretHash));
      dispatch(addSwapItem({
        amountBase: baseQuantity,
        amountQuote: quoteQuantity,
        swapId: data.swap.id,
        swapHash: data.swap.secretHash,
        secretSeekerId: data.swap.secretSeeker.id,
        secretHolderId: data.swap.secretHolder.id,
        secret: data.swapSecret
      }));
    })
    .catch(err => console.log(err))
    
  }

  const onViewHistory = () => {
    navigate('/history');
  };

  return (
    <Card centered>
      <Card.Content>
        <Card.Header>
          Initate New Swap
        </Card.Header><br />
        <Form>
        <Form.Group  widths='equal'>
          <Form.Field>
            <label>Base Quantity: 
            <input type='number' value={baseQuantity} onChange={(evt) => setBaseQuantity(evt.target.value)}/></label>
          </Form.Field>
          <Form.Field>
            <label>Quote Quantity: 
            <input type='number' value={quoteQuantity} onChange={(evt) => setQuoteQuantity(evt.target.value)}/></label>
          </Form.Field>
        </Form.Group>
        <p><Button primary onClick={onCreateSwap}>Create Swap</Button></p>
        <p><Button primary onClick={onViewHistory}>Swap History</Button></p>
        </Form>
      </Card.Content>
    </Card>
  );
}
