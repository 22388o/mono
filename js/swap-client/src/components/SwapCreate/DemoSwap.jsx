import React from 'react';
import { Button } from 'semantic-ui-react';

export const DemoSwap = ({ mockSwap }) => {
  return <>
    <p>Demo Swap</p>
    <Button circular secondary className='gradient-btn w-100 h-3' onClick={e => mockSwap({
      side: 'ask',
      baseAsset: 'BTC',
      baseNetwork: 'lightning.btc',
      baseQuantity: 0.001,
      quoteAsset: 'ETH',
      quoteNetwork: 'eth-l2.eth',
      quoteQuantity: 0.0000000000001})}>Swap 0.001 BTC for 0.0000000000001 ETH</Button>
    <Button circular secondary className='gradient-btn w-100 h-3' onClick={e => mockSwap({
      side: 'bid',
      baseAsset: 'ETH',
      baseNetwork: 'eth-l2.eth',
      baseQuantity: 0.0000000000001,
      quoteAsset: 'BTC',
      quoteNetwork: 'lightning.btc',
      quoteQuantity: 0.001})}>Swap 0.0000000000001 ETH for 0.001 BTC</Button>
  </>;
}