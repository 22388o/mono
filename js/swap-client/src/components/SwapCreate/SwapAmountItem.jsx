import React from 'react';
import { Dropdown, Form } from 'semantic-ui-react';
import styles from '../styles/SwapCreate.module.css';

const friendOptions = [
  {
    key: 'BTC',
    text: 'BTC',
    value: 'BTC',
    image: { avatar: true, src: 'https://github.com/dapphub/trustwallet-assets/blob/master/blockchains/bitcoin/info/logo.png?raw=true' },
  },
  {
    key: 'ETH',
    text: 'ETH',
    value: 'ETH',
    image: { avatar: true, src: 'https://github.com/dapphub/trustwallet-assets/blob/master/blockchains/ethereum/info/logo.png?raw=true' },
  }
]

export const SwapAmountItem = ({coinType, amount, className, onAmountChange, unitPrice, onCoinTypeChange, availQty, limitOrder = true}) => {
  const links = {
    'btc': "https://github.com/dapphub/trustwallet-assets/blob/master/blockchains/bitcoin/info/logo.png?raw=true",
    'eth': "https://github.com/dapphub/trustwallet-assets/blob/master/blockchains/ethereum/info/logo.png?raw=true"
  };

  const onKeyDown = (e) => {
    if(e.keyCode === 109) {
      e.preventDefault();
      return;
    }
  }

  return <Form.Group widths='equal' className={className}>
    <Form.Field className={styles.swapAmountInput}>
      { limitOrder ? 
          <input placeholder={availQty} type='number' value={(amount===0)?'':amount} onChange={onAmountChange} onKeyDown={onKeyDown}/>
        :
          <input placeholder={availQty} type='number' value={(amount===0)?'':amount} onChange={onAmountChange} onKeyDown={onKeyDown}/>
      }
      { unitPrice * amount > 0 ? <p className={styles.price}>${unitPrice * amount}</p> : ''}
    </Form.Field>
    <Form.Field className={styles.coinType}>
      <Dropdown
        className={styles.swapCoinSelect}
        floating
        fluid
        labeled
        button
        value={coinType}
        options={friendOptions}
        onChange={onCoinTypeChange}
      />
    </Form.Field>
  </Form.Group>
};