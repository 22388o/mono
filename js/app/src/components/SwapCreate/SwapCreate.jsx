import React, { useCallback, useEffect, useState, useSyncExternalStore } from "react";
import { Box, Grid, Stack, Button, IconButton, Divider } from "@mui/material";
import { getBTCPrice, getETHPrice } from "../../utils/apis";
import styles from '../../styles/SwapCreate.module.css';
import { SwapAmountItem } from "./SwapAmountItem";
import { hashSecret, log } from "../../utils/helpers";
import SettingsEthernetIcon from '@mui/icons-material/SettingsEthernet';
import { walletStore } from "../../syncstore/walletstore";
import { activitiesStore } from "../../syncstore/activitiesstore";
import { toast } from "react-toastify";

/**
 * Swap Form Component which handles swap
 */
export const SwapCreate = () => {
  const globalWallet = useSyncExternalStore(walletStore.subscribe, () => walletStore.currentState);
  const ASSET_TYPES = globalWallet.assets;

  const [baseQuantity, setBaseQuantity] = useState(0);
  const [quoteQuantity, setQuoteQuantity] = useState(0);
  const [baseAsset, setBaseAsset] = useState(0);
  const [quoteAsset, setQuoteAsset] = useState(1);
  const [curPrices, setCurPrices] = useState({
    BTC: 0,
    ETH: 0,
    fetching: true
  });

  const node = globalWallet.assets[0];
  const wallet = globalWallet.assets[1];

  useEffect(() => {
    const core = async () => {
      const btc = await getBTCPrice();
      const eth = await getETHPrice();
      setCurPrices({
        BTC: btc,
        ETH: eth,
        fetching: false
      });
    };
    core();
  }, []);

  /**
   * Handler for input base quantity
   */
  const onInputBaseQuantity = useCallback((e) => {
    if(e.target.value < 0) return;
    setBaseQuantity(e.target.value);
  }, [curPrices, ASSET_TYPES, baseAsset, quoteAsset]);

  /**
   * Handler for input quote quantity
   */
  const onInputQuoteQuantity = useCallback((e) => {
    if(e.target.value < 0) return;
    setQuoteQuantity(e.target.value);
  }, [curPrices, ASSET_TYPES, quoteAsset, baseAsset]);

  /**
   * Handles swap order
   */
  const onOrderSwap = useCallback(async (order) => {
    const randomValues = crypto.getRandomValues(new Uint8Array(32))
    const secretHex = [...randomValues]
      .map(byte => byte.toString(16).padStart(2, '0'))
      .join('')
    const secretHash = await hashSecret(randomValues);

    if(ASSET_TYPES[baseAsset].balance < baseQuantity) {
      toast.error(
        "Error",
        {
          theme: 'colored',
          autoClose: 1000
        }
      )
      return;
    }

    await thenOrderSwap(order, secretHex, secretHash);
  }, [crypto, ASSET_TYPES, baseAsset, baseQuantity, quoteQuantity, quoteAsset]);

  /**
   * Process Order
   * @param {Object} order 
   * @param {string} secret 
   * @param {string} secretHash 
   */
  const thenOrderSwap = async (order, secret, secretHash) => {
    const ask = order.side=='ask';
    const baseA = order.baseAsset ? order.baseAsset : ASSET_TYPES[baseAsset].type;
    const quoteA = order.quoteAsset ? order.quoteAsset : ASSET_TYPES[quoteAsset].type;
    const baseQty = order.baseQuantity ? order.baseQuantity : baseQuantity;
    const quoteQty = order.quoteQuantity ? order.quoteQuantity : quoteQuantity;
    const baseNet= order.baseNetwork, quoteNet = order.quoteNetwork;
    const baseO = { asset: baseA, network: baseNet, quantity: baseQty },
        quoteO = { asset: quoteA,  network: quoteNet, quantity: quoteQty };

    let args = ask ?  { // if order is an ask, bitcoin as base
      base: baseO,
      quote: quoteO
    } : {
      base: quoteO,
      quote: baseO
    };

    let bai = ASSET_TYPES.findIndex(asset => asset.type === args.base.asset);
    let qai = ASSET_TYPES.findIndex(asset => asset.type === args.quote.asset);

    const request = {
      uid: 'alice', // Change: user.user.id,
      side: order.side,
      hash: secretHash,
      baseAsset: args.base.asset.split('-')[0],
      baseNetwork: args.base.network,
      baseQuantity: (args.base.asset.split('-')[0] === 'BTCORD' || args.quote.asset.split('-')[0] === 'BTCORD') ? 4000 : Math.round(args.base.quantity * ASSET_TYPES[bai].rate),
      baseInfo: ASSET_TYPES[baseAsset].info,
      quoteAsset: args.quote.asset.split('-')[0],
      quoteNetwork: args.quote.network,
      quoteQuantity: Math.round(args.quote.quantity * ASSET_TYPES[qai].rate),
      quoteInfo: ASSET_TYPES[quoteAsset].info
    };
    
    const curDate = new Date();
    const date = {
      year: curDate.getFullYear(),
      month: curDate.getMonth(),
      day: curDate.getDate()
    };

    const baseQ = {
          asset: request.baseAsset,
          network: order.baseNetwork,
          quantity: request.baseQuantity / ASSET_TYPES[bai].rate
        }, quoteQ = {
          asset: request.quoteAsset,
          network: order.quoteNetwork,
          quantity: request.quoteQuantity / ASSET_TYPES[qai].rate
        };

    args = ask ?  { // if order is an ask, bitcoin as base
      base: baseQ,
      quote: quoteQ
    } : {
      base: quoteQ,
      quote: baseQ
    };

    activitiesStore.dispatch({ type: 'ADD_SWAP_ITEM', payload: {
      key: request.id,
      orderId: request.id,
      ts: request.ts,
      uid: request.uid,
      type: request.type,
      side: request.side,
      secret: secret,
      secretHash,
      hash: request.hash,
      baseAsset: args.base.asset.split('-')[0],
      baseQuantity: args.base.quantity,
      baseNetwork: args.base.network,
      baseInfo: ASSET_TYPES[baseAsset].info,
      quoteAsset: args.quote.asset.split('-')[0],
      quoteNetwork: args.quote.network,
      quoteQuantity: args.quote.quantity,
      quoteInfo: ASSET_TYPES[quoteAsset].info,
      ordinalLocation: args.ordinalLocation,
      status: 0,
      createdDate: date
    } })

    walletStore.dispatch({ type: 'REMOVE_BALANCE_ON_SWAP_ORDER', payload: {asset: baseAsset, qty: baseQuantity} })
    setBaseQuantity(ASSET_TYPES[baseAsset].isNFT ? 1 : 0);
    setQuoteQuantity(ASSET_TYPES[quoteAsset].isNFT ? 1 : 0);
  }

  /**
   * Handler to exchange base & quote coin type
   */
  const onExchangeCoinType = useCallback(() => {
    const tBase = baseQuantity, tQuote = quoteQuantity;
    const aBase = baseAsset, aQuote = quoteAsset;
    setBaseAsset(aQuote);setQuoteAsset(aBase);
    setBaseQuantity(tQuote); setQuoteQuantity(tBase);
  }, [baseQuantity, quoteQuantity, baseAsset, quoteAsset]);

  return (
    <Box className={styles.SwapCreateContainer}>
      <Stack sx={{alignItems: 'center'}}>
        <Grid container height={35} sx={{marginBottom:'1em'}}>
          <Grid item xs={4} textAlign='left' style={{display:'flex', alignItems:'center'}}><h3>Swap</h3></Grid>
        </Grid>
        <Grid className={styles.swapExCont}>
          <SwapAmountItem
            assetId={baseAsset}
            unitPrice={curPrices[ASSET_TYPES[baseAsset].type]}
            amount={baseQuantity}
            availQty={ASSET_TYPES[baseAsset].type === 'BTC' ? node.balance : wallet.balance}
            onAmountChange={onInputBaseQuantity}
            onCoinTypeChange={(asset) => {coinTypeChanged(true, asset);}}
          />
          <Divider style={{borderColor:'#202020',margin: '0.5em -1em 0 -1em'}} />
          <IconButton className={`${styles.exchange} exchange`} onClick={onExchangeCoinType}><SettingsEthernetIcon /></IconButton>
          <SwapAmountItem
            className='mt-m1 mb-0'
            assetId={quoteAsset}
            unitPrice={curPrices[ASSET_TYPES[quoteAsset].type]}
            amount={quoteQuantity}
            availQty={ASSET_TYPES[baseAsset].type === 'ETH' ? node.balance : wallet.balance}
            onAmountChange={onInputQuoteQuantity}
            onCoinTypeChange={(asset) => {coinTypeChanged(false, asset);}}
            />
        </Grid>
        <Grid sx={{width:'100%', marginTop:'1em'}}>
          { 
            ((ASSET_TYPES[baseAsset].isNFT || baseQuantity) && (ASSET_TYPES[quoteAsset].isNFT || quoteQuantity)
                ? <>
                    { (ASSET_TYPES[baseAsset].isNFT==false && ASSET_TYPES[quoteAsset].isNFT==false) ?  <p className={styles.prices}>{ curPrices.fetching ? 'Loading' : `1 ${ASSET_TYPES[baseAsset].type} = ${Number(curPrices[ASSET_TYPES[baseAsset].type] / curPrices[ASSET_TYPES[quoteAsset].type]).toFixed(6)} ${ASSET_TYPES[quoteAsset].type}` }</p> :
                    <p></p>}
                    <Button circular="true" secondary="true" className='gradient-btn w-100 h-3' onClick={e => onOrderSwap({
                      side: (
                      (ASSET_TYPES[quoteAsset].type.split('-')[0] !== 'BTCORD' && (ASSET_TYPES[baseAsset].type == 'BTC' || ASSET_TYPES[baseAsset].type.split('-')[0] == 'BTCORD' || ASSET_TYPES[baseAsset].isNFT)) ? 'ask' : 'bid'),
                      baseNetwork: ASSET_TYPES[baseAsset].network,
                      quoteNetwork: ASSET_TYPES[quoteAsset].network,
                      ordinalLocation:  (
                        (ASSET_TYPES[baseAsset].isNFT || ASSET_TYPES[quoteAsset].isNFT )?
                          (ASSET_TYPES[baseAsset].isNFT ?
                            ASSET_TYPES[baseAsset].info.location :
                            ASSET_TYPES[quoteAsset].info.location) :
                          false )})}>Swap</Button>
                  </>
                : <Button circular="true" secondary="true" className='w-100 h-3 gradient-btn-disabled' disabled>Enter Amounts to Swap</Button> )
          }
        </Grid>
      </Stack>
    </Box>
  );
}
