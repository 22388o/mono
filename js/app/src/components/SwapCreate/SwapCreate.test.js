import { fireEvent, render } from '@testing-library/react';
import { SwapCreate } from './SwapCreate';

Object.defineProperty(global.self, "crypto", {
  value: {
    getRandomValues: (arr) => "123123123",
    subtle: {
      digest: (algorithm, data) => {
        return new Promise((resolve, reject) =>
          resolve(
            "123123"
          )
        );
      },
    },
  },
});

global.fetch = jest.fn(() =>
  Promise.resolve({
    json: () => Promise.resolve({ data: { amount: 12 } }),
  })
);

describe('SwapCreate test', () => {
  it('should work', () => {

    const { container } = render(
      <SwapCreate />
    );
    expect(container).toMatchSnapshot();

    const exchangeBtn = container.querySelector('.exchange');
    fireEvent.click(exchangeBtn);
    expect(container).toMatchSnapshot();
  })

  it('should call handler when input', () => {

    const { container } = render(
      <SwapCreate />
    );

    const inputs = container.querySelectorAll('.qty-input');
    fireEvent.change(inputs[0], { target: {value: "1235"} });
    expect(inputs[0].value).toBe("1235");
    fireEvent.change(inputs[1], { target: {value: "123"} });
    expect(inputs[1].value).toBe("123");
    const swapBtn = container.querySelector('.gradient-btn');
    fireEvent.click(swapBtn);

    fireEvent.change(inputs[0], { target: {value: "12"} });
    expect(inputs[0].value).toBe("12");
    fireEvent.click(swapBtn);
  })
  afterAll(() => jest.resetModules())
})
