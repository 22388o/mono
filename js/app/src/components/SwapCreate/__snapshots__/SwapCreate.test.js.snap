// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwapCreate test should work 1`] = `
<div>
  <div
    class="MuiBox-root css-0"
  >
    <div
      class="MuiStack-root css-163ljxj-MuiStack-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container css-1o9533o-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-z58hwb-MuiGrid-root"
          style="display: flex; align-items: center;"
        >
          <h3>
            Swap
          </h3>
        </div>
      </div>
      <div
        class="MuiGrid-root css-vj1n65-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container css-1tj2tdk-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-7 css-1smoqvn-MuiGrid-root"
          >
            <input
              class="undefined qty-input"
              placeholder="1000"
              type="number"
              value=""
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-14m1l2r-MuiGrid-root"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium undefined coin-select css-1e6y48t-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              <img
                src="https://github.com/dapphub/trustwallet-assets/blob/master/blockchains/bitcoin/info/logo.png?raw=true"
              />
              BTC
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
          style="border-color: #202020; margin: 0.5em -1em 0px -1em;"
        />
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium undefined exchange css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="SettingsEthernetIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7.77 6.76 6.23 5.48.82 12l5.41 6.52 1.54-1.28L3.42 12l4.35-5.24zM7 13h2v-2H7v2zm10-2h-2v2h2v-2zm-6 2h2v-2h-2v2zm6.77-7.52-1.54 1.28L20.58 12l-4.35 5.24 1.54 1.28L23.18 12l-5.41-6.52z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <div
          class="MuiGrid-root MuiGrid-container mt-m1 mb-0 css-1tj2tdk-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-7 css-1smoqvn-MuiGrid-root"
          >
            <input
              class="undefined qty-input"
              placeholder="1000"
              type="number"
              value=""
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-14m1l2r-MuiGrid-root"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium undefined coin-select css-1e6y48t-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              <img
                src="https://github.com/dapphub/trustwallet-assets/blob/master/blockchains/ethereum/info/logo.png?raw=true"
              />
              ETH
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root css-e54n9e-MuiGrid-root"
      >
        <button
          circular="true"
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium Mui-disabled MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium w-100 h-3 gradient-btn-disabled css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          disabled=""
          secondary="true"
          tabindex="-1"
          type="button"
        >
          Enter Amounts to Swap
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`SwapCreate test should work 2`] = `
<div>
  <div
    class="MuiBox-root css-0"
  >
    <div
      class="MuiStack-root css-163ljxj-MuiStack-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container css-1o9533o-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-z58hwb-MuiGrid-root"
          style="display: flex; align-items: center;"
        >
          <h3>
            Swap
          </h3>
        </div>
      </div>
      <div
        class="MuiGrid-root css-vj1n65-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container css-1tj2tdk-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-7 css-1smoqvn-MuiGrid-root"
          >
            <input
              class="undefined qty-input"
              placeholder="1000"
              type="number"
              value=""
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-14m1l2r-MuiGrid-root"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium undefined coin-select css-1e6y48t-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              <img
                src="https://github.com/dapphub/trustwallet-assets/blob/master/blockchains/ethereum/info/logo.png?raw=true"
              />
              ETH
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
          style="border-color: #202020; margin: 0.5em -1em 0px -1em;"
        />
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium undefined exchange css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="SettingsEthernetIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7.77 6.76 6.23 5.48.82 12l5.41 6.52 1.54-1.28L3.42 12l4.35-5.24zM7 13h2v-2H7v2zm10-2h-2v2h2v-2zm-6 2h2v-2h-2v2zm6.77-7.52-1.54 1.28L20.58 12l-4.35 5.24 1.54 1.28L23.18 12l-5.41-6.52z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <div
          class="MuiGrid-root MuiGrid-container mt-m1 mb-0 css-1tj2tdk-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-7 css-1smoqvn-MuiGrid-root"
          >
            <input
              class="undefined qty-input"
              placeholder="1000"
              type="number"
              value=""
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-14m1l2r-MuiGrid-root"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium undefined coin-select css-1e6y48t-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              <img
                src="https://github.com/dapphub/trustwallet-assets/blob/master/blockchains/bitcoin/info/logo.png?raw=true"
              />
              BTC
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root css-e54n9e-MuiGrid-root"
      >
        <button
          circular="true"
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium Mui-disabled MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium w-100 h-3 gradient-btn-disabled css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          disabled=""
          secondary="true"
          tabindex="-1"
          type="button"
        >
          Enter Amounts to Swap
        </button>
      </div>
    </div>
  </div>
</div>
`;
